File: 251001/Brutskaya/disCo/publisher/src/main/java/com/example/publisher/model/Issue.java
package com.example.publisher.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.Instant;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@Table(name = "tbl_issue")
public class Issue extends BaseEntity{

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(length = 64, nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    private Instant created;

    private Instant modified;

    @ManyToMany
    @JoinTable(
            name = "tbl_issue_label",
            joinColumns = @JoinColumn(name = "issue_id"),
            inverseJoinColumns = @JoinColumn(name = "label_id")
    )
    private Set<Label> labels;
}

File: 251003/Pankratiev/rest/publisher/src/main/kotlin/bsuir/dc/publisher/entity/Issue.kt
package bsuir.dc.publisher.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "tbl_issue", schema = "public")
data class Issue(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "writerId", nullable = false)
    val writer: Writer = Writer(),

    @Column(nullable = false, unique = true, length = 64)
    val title: String = "",

    @Column(nullable = false, length = 2048)
    val content: String = "",

    @Column(nullable = false)
    val created: LocalDateTime = LocalDateTime.now(),

    @Column(nullable = false)
    var modified: LocalDateTime = LocalDateTime.now(),

    @OneToMany(mappedBy = "issue", cascade = [CascadeType.ALL], orphanRemoval = true)
    val posts: MutableList<Post> = mutableListOf(),

    @ManyToMany(cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    @JoinTable(
        name = "tbl_issue_label",
        schema = "public",
        joinColumns = [JoinColumn(name = "issueId")],
        inverseJoinColumns = [JoinColumn(name = "labelId")]
    )
    val labels: MutableSet<Label> = mutableSetOf()
)
